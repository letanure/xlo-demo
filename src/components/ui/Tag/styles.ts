import styled, { css } from 'styled-components'

import { TagProps } from '.'

export const Wrapper = styled.span<TagProps>`
  ${({ theme }) => css`
    align-items: center;
    background: #ffce32;
    border-radius: 4px;
    display: flex;
    display: inline-block;
    font-family: ${theme.font.family.heading.name};
    font-size: 12px;
    font-style: normal;
    font-weight: 500;
    letter-spacing: 0.8px;
    line-height: 16px;
    padding: 4px 8px;
    text-transform: uppercase;
  `}
`
