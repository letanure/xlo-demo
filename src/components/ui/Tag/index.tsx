import * as S from './styles'

export type TagProps = {
  text: string | number
}

const Tag = ({ text }: TagProps) => <S.Wrapper text={text}>{text}</S.Wrapper>

export default Tag
