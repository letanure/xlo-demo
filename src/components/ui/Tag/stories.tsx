import { ComponentStory, ComponentMeta } from '@storybook/react'

import Tag from '.'

export default {
  title: 'UI/Tag',
  component: Tag,
  argTypes: {
    text: { control: 'text' }
  }
} as ComponentMeta<typeof Tag>

const Template: ComponentStory<typeof Tag> = (args) => <Tag {...args} />

export const Basic = Template.bind({})
Basic.args = {
  text: 'tag'
}
