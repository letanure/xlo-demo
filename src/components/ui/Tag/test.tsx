import { render, screen } from '@testing-library/react'

import Tag from '.'

describe('<Tag />', () => {
  it.skip('should render something', () => {
    const { container } = render(<Tag text="tag" />)

    expect(screen.getByRole('heading', { name: /Tag/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
