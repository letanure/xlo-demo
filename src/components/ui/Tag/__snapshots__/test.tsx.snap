// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should render something 1`] = `
.c0 {
  content: 'default';
}

<div
  class="c0"
>
  <h1>
    Tag with 
    default
  </h1>
</div>
`
